lib/transport/abstract-transport.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var util = require("util");
+var EventEmitter = require('events');
+
+var AbstractTransport = function () {
+	if (this.constructor === AbstractTransport) {
+		throwBecauseAbstract();
```

Comment:
well that's one of the checks you can make during registration. I think you'll find it a far more agreeable design to not export the `AbstractTransport`, otherwise you'll have to add a circular dependency in your custom transport. This class can be used internally as the base class of the known transports for sure though